.github/workflows/flags.yml

```diff
@@ -70,7 +70,7 @@ jobs:
           os:  ${{ runner.os }}
 
       - name: Build `hls-graph` with flags
-        run: cabal v2-build hls-graph --flags="embed-files stm-stats"
+        run: cabal v2-build hls-graph --flags="embed-files stm-stats pedantic"
```

Comment:
I _think_ this should be implied by the later build with pedantic?

Reply:
Indeed, that's what I expected too (that when you change flags cabal rebuilds all the libraries affected).

But the how do you explain the fact that this flags job passes on master with all those warnings present?

Running these 3 cabal builds locally (after cabal clean) I see that modules from hls-graph are compiled in the first two and the third (`cabal v2-build --flags="pedantic"`) seems to be using hls-graph compiled in the first 2 runs:

```
$ cabal v2-build --flags="pedantic"
...
In order, the following will be built (use -v for more details):
 - ghcide-2.6.0.0 (exe:ghcide-test-preprocessor) (configuration changed)
 - hls-graph-2.6.0.0 (lib) (configuration changed)
 - shake-bench-0.2.0.0 (lib) (first run)
...
Preprocessing executable 'ghcide-test-preprocessor' for ghcide-2.6.0.0..
Building executable 'ghcide-test-preprocessor' for ghcide-2.6.0.0..
Preprocessing library for hls-graph-2.6.0.0..
Building library for hls-graph-2.6.0.0..                                   <<-- the output suggests the individual modules are no longer compiled (it's probably using artifacts from previous 2 builds???
Preprocessing library for shake-bench-0.2.0.0..
Building library for shake-bench-0.2.0.0..
```

Reply:
Weird. I wonder if we could simplify this by setting up a `cabal.project.local` as for the test options and then just doing a simple `cabal build all`?